src/pe_source/data/shodan/shodan_search.py

```diff
@@ -0,0 +1,485 @@
+
+        failed = search_shodan(
+            thread_name, ips, api, start, end, org_uid, org_name, failed
+        )
+
+    if len(failed) > 0:
+        logging.critical("{} Failures: {}".format(thread_name, failed))
+
+
+def get_dates():
+    """Get dates for the query."""
+    now = datetime.datetime.now()
+    days_back = datetime.timedelta(days=30)
+    days_forward = datetime.timedelta(days=1)
+    start = now - days_back
+    end = now + days_forward
+    start_time = time_to_utc(start)
+    end_time = time_to_utc(end)
+    return start_time, end_time
+
+
+def time_to_utc(in_time):
+    """Convert time to UTC."""
+    # If time does not have timezone info, assume it is local
+    if in_time.tzinfo is None:
+        local_tz = datetime.datetime.now().astimezone().tzinfo
+        in_time = in_time.replace(tzinfo=local_tz)
+    utc_time = in_time.astimezone(datetime.timezone.utc)
+    return utc_time
+
+
+def search_circl(cve):
+    """Fetch CVE info from Circl."""
+    re = requests.get(f"https://cve.circl.lu/api/cve/{cve}")
+    return re
+
+
+def search_shodan(thread_name, ips, api, start, end, org_uid, org_name, failed):
+    """Search IPs in the Shodan API."""
+    # Initialize lists to store Shodan results
+    data = []
+    risk_data = []
+    vuln_data = []
+
+    # Build dictionaries for naming conventions and definitions
+    risky_ports, name_dict, risk_dict, av_dict, ac_dict, ci_dict = get_shodan_dicts()
+
+    # Break up IPs into chunks of 100
+    # Throws flake8 E203 error which is ignored in .flake8
+    ip_chunks = [ips[i : i + 100] for i in range(0, len(ips), 100)]
+    tot_ips = len(ips)
+    tot = len(ip_chunks)
+    logging.info(
+        "{} Split {} IPs into {} chunks - {}".format(
+            thread_name, tot_ips, tot, org_name
+        )
+    )
+
+    # Loop through chunks and search Shodan
+    for i, ip_chunk in enumerate(ip_chunks):
+        try_count = 1
+        while try_count < 7:
+            try:
+                results = api.host(ip_chunk, history=True)
+                for r in results:
+                    for d in r["data"]:
+                        # Convert Shodan date string to UTC datetime
+                        shodan_datetime = datetime.datetime.strptime(
+                            d["timestamp"], "%Y-%m-%dT%H:%M:%S.%f"
+                        )
+                        shodan_utc = time_to_utc(shodan_datetime)
+                        # Only include results in the timeframe
+                        if shodan_utc > start and shodan_utc < end:
+                            prod = d.get("product", None)
+                            serv = d.get("http", {}).get("server")
+                            asn = d.get("ASN", None)
+                            vulns = d.get("vulns", None)
+                            if vulns is not None:
+                                cves = list(vulns.keys())
+                                unverified = []
+                                for cve in cves:
```

Comment:
Remove single-use variable:
```suggestion
                                unverified = []
                                for cve in list(vulns.keys()):
```